config/config.go

```diff
@@ -58,18 +59,41 @@ type File struct {
 
 // New returns a new configuration
 func New() repository.Config {
-	return &File{}
+	return &Configuration{}
 }
 
 // LoadConfig handles viper under the hood
-func (cfg *File) LoadConfig() {
-	viper.SetConfigName("config") // name of config file (without extension)
-	viper.SetConfigType("yaml")   // REQUIRED if the config file does not have the extension in the name
-	viper.AddConfigPath(".")      // optionally look for config in the working directory
-	viper.SetDefault("sources.directories", "./games")
+func (cfg *Configuration) LoadConfig() {
+	viper.SetConfigName("config")     // name of config file (without extension)
+	viper.SetConfigType("yaml")       // REQUIRED if the config file does not have the extension in the name
+	viper.AddConfigPath(".")          // optionally look for config in the working directory
+	viper.SetTypeByDefaultValue(true) // Allows []string to be parsed from Env Vars
+
+	viper.SetDefault("host", "tinshop.example.com")
+	viper.SetDefault("protocol", "http")
```

Comment:
Is function `ComputeDefaultValues`, `http` is assumed if no protocol, so default value is not needed

Reply:
I think this is a matter of preference, so I will make my case but accept your decision.
`viper` needs a default set (could be `nil`) in order to assign ENV vars to configuration keys.
For the sake of standardization, I believe this could stay here, and the `ComputeDefaultValues` should test for invalid values and force them to be the default, instead of being the default section “setter”. This way, all the default values are in one place. What are your thoughts?